test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts

```diff
@@ -0,0 +1,85 @@
+
+import BigNumber from 'bignumber.js';
+import sinon from 'sinon';
+import * as Web3Utils from 'web3-utils';
+
+import RedeemIntentDeclaredHandler from '../../../../src/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler';
+import Message from '../../../../src/models/Message';
+import {
+  MessageDirection, MessageRepository, MessageStatus, MessageType,
+} from '../../../../src/repositories/MessageRepository';
+import assert from '../../../test_utils/assert';
+import SpyAssert from '../../../test_utils/SpyAssert';
+
+describe('RedeemIntentDeclaredHandler.persist()', (): void => {
+  const transactions = [{
+    _messageHash: Web3Utils.keccak256('1'),
+    _redeemer: '0x0000000000000000000000000000000000000001',
+    _redeemerNonce: '1',
+    _beneficiary: '0x0000000000000000000000000000000000000002',
+    _amount: '100',
+    contractAddress: '0x0000000000000000000000000000000000000002',
+    blockNumber: '10',
+  }];
+
+  it('should change message state to source declared', async (): Promise<void> => {
```

Comment:
One more possible test `should not change message state if source status is already declared`

Reply:
The test is already there. See [line no 57](https://github.com/mosaicdao/facilitator/blob/a25ca891b3aeab57e43db13a18bc1c3dc4af6fe7/test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts#L57)

Reply:
Added test.